src/app/(organization)/credits/costs/page.tsx

```diff
@@ -1,42 +1,40 @@
 'use client';
+import { useQuery } from '@apollo/client';
 import { createColumnHelper } from '@tanstack/react-table';
 import GridTable from '../../../../components/GridTable';
-import Table from '../../../../components/Table';
-
-interface CreditBlockchain {
-  cost: number;
-}
+import { GetCreditSheet } from '../../../../queries/credits.graphql';
+import { ActionCost, Blockchain, BlockchainCost } from '../../../../graphql.types';
 
 interface CreditLineItem {
   action: string;
-  blockchains: CreditBlockchain[];
+  solana: number;
+  polygon: number;
+}
+
+interface GetCreditSheetData {
+  creditSheet: ActionCost[];
 }
+
 export default function CostPage() {
+  const creditSheetQuery = useQuery<GetCreditSheetData>(GetCreditSheet);
+  console.log('creditSheet', creditSheetQuery.data);
+  const creditSheet = creditSheetQuery.data?.creditSheet;
+  const loading = creditSheetQuery.loading;
+  const data: CreditLineItem[] = [];
+  creditSheet?.map((ac: ActionCost) => {
+    data.push({
+      action: ac.action as string,
+      solana: ac.blockchains.filter((bc: BlockchainCost) => bc.blockchain === Blockchain.Solana)[0]
+        .credits,
+      polygon: ac.blockchains.filter(
+        (bc: BlockchainCost) => bc.blockchain === Blockchain.Polygon
+      )[0].credits,
+    });
```

Comment:
A reduce is good here to convert the list of blockchain to keys on a new struct.

Reply:
Done